### DISCUSSION TITLE
Typing a store wrapping a composable

### DISCUSSION THREAD

**Discussion by PaulHax** [vuejs/pinia#2258] [Help and Questions]
I've got a composable, `useAnnotationTool`, that is wrapped by 2 "children" stores, `rulerTool` and `rectangleTool`.  The stores mostly re-expose the refs and functions implemented in `useAnnotationTool`.  

I'ed like to pass ether the `rulerTool` or `rectangleTool` stores as a component property on a `AnnotationTool` component.  How do you create a Typescript type for "a store that wraps the `useAnnotationTool`"?

This does not seem to work:
```
type UseAnnotationTool = ReturnType<typeof useAnnotationTool>
type AnnotationToolStore = StoreState<UseAnnotationTool> & StoreActions<UseAnnotationTool>
```

The code I'm working on:

useAnnotationTool: https://github.com/Kitware/VolView/blob/b4628956a90c967bd1756da4745b9131a743f58a/src/store/tools/useAnnotationTool.ts#L31

rectangles store:
https://github.com/Kitware/VolView/blob/b4628956a90c967bd1756da4745b9131a743f58a/src/store/tools/rectangles.ts#L20-L27

Maybe there is a better way to approach this sharing of logic across "sibling" stores?  Instantiate a `annotationTool` store inside each `rectangles` and `rulers` stores?

**Comment by posva** [comment]
The one you mentioned should be the one working:

```ts
type AnnotationToolStore = ReturnType<typeof useAnnotationTool>
```
